File: src/main/java/com/back/domain/member/memberGame/repository/MemberGameRepository.java
package com.back.domain.member.memberGame.repository;

import com.back.domain.member.memberGame.StatusEnum;
import com.back.domain.member.memberGame.entity.MemberGame;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface MemberGameRepository extends JpaRepository<MemberGame, Integer> {
    Optional<MemberGame> findByMemberIdAndGameId(int memberId, Long gameId);

    Page<MemberGame> findByMemberId(int memberId, Pageable pageable);

    Page<MemberGame> findByMemberIdAndStatus(int memberId, StatusEnum status, Pageable pageable);

    Page<MemberGame> findByMemberIdAndPlatformIdIn(int memberId, List<Long> platformIds, Pageable pageable);

    Page<MemberGame> findByMemberIdAndStatusAndPlatformIdIn(int memberId, StatusEnum status, List<Long> platformIds, Pageable pageable);

    List<MemberGame> findAllByMemberId(int memberId);
}

File: src/main/kotlin/com/back/domain/member/memberGame/repository/MemberGameRepository.kt
package com.back.domain.member.memberGame.repository

import com.back.domain.member.memberGame.StatusEnum
import com.back.domain.member.memberGame.entity.MemberGame
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface MemberGameRepository : JpaRepository<MemberGame, Int> {
    fun findByMemberIdAndGameId(
        memberId: Int,
        gameId: Int,
    ): MemberGame?

    fun findByMemberId(
        memberId: Int,
        pageable: Pageable,
    ): Page<MemberGame>

    fun findByMemberIdAndStatus(
        memberId: Int,
        status: StatusEnum,
        pageable: Pageable,
    ): Page<MemberGame>

    fun findByMemberIdAndPlatformIdIn(
        memberId: Int,
        platformIds: List<Long>,
        pageable: Pageable,
    ): Page<MemberGame>

    fun findByMemberIdAndStatusAndPlatformIdIn(
        memberId: Int,
        status: StatusEnum,
        platformIds: List<Long>,
        pageable: Pageable,
    ): Page<MemberGame>

    fun findAllByMemberId(memberId: Int): List<MemberGame>
}
